apps/ledger-live-desktop/src/renderer/families/evm/StakeFlowModal/index.tsx

```diff
@@ -1,23 +1,172 @@
+            height: ${SHADOW_HEIGHT}px;
+            background: linear-gradient(to bottom, ${theme.colors.background.main}, transparent);
+            z-index: 1;
+            pointer-events: none;
+          }
+        `
+       : ""
+   }
+`,
+);
+
+const Footer = styled(Box)<IsScrollable>(
+  ({ isScrollable, theme }) => `
+    position: relative;
+
+    ${
+      isScrollable
+        ? `
+          ::before {
+              content: "";
+              position: absolute;
+              bottom: 100%;
+              left: 0;
+              right: ${SCROLL_WIDTH}px;
+              height: ${SHADOW_HEIGHT}px;
+              background: linear-gradient(to top, ${theme.colors.background.main}, transparent);
+              z-index: 1;
+              pointer-events: none;
+            }
+        `
+        : ""
+    }
+
+  `,
+);
+
+// Comparison fns for sorting providers by minimum ETH required
+const ascending = (a: EthStakingProvider, b: EthStakingProvider) => (a?.min ?? 0) - (b?.min ?? 0);
+const descending = (a: EthStakingProvider, b: EthStakingProvider) => (b?.min ?? 0) - (a?.min ?? 0);
+
+type Option = EthStakingProviderCategory | "all";
+const OPTION_VALUES: Option[] = ["all", "liquid", "protocol", "pooling", "restaking"] as const;
+
+export interface Props {
   account: Account;
-  singleProviderRedirectMode?: boolean;
   /** Analytics source */
   source?: string;
   hasCheckbox?: boolean;
-};
+}
+
+const MODAL_WIDTH = 500;
+
+export const StakeModal = ({ account, source }: Props) => {
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const store = useStore();
+
+  const hasMinValidatorEth = account.spendableBalance.isGreaterThan(ETH_LIMIT);
 
-const StakingModal = ({ account, hasCheckbox, singleProviderRedirectMode, source }: Props) => {
   const ethStakingProviders = useFeature("ethStakingProviders");
   const providers = ethStakingProviders?.params?.listProvider;
 
+  const [selected, setSelected] = useState<Option>("all");
+
+  const formattedProviders = useMemo(
+    () =>
+      (providers ?? [])
+        .filter(x => !x.disabled && (selected === "all" || selected === x.category))
+        .sort(hasMinValidatorEth ? descending : ascending),
+    [providers, hasMinValidatorEth, selected],
+  );
+
+  const scrollableContainerRef = useRef<HTMLDivElement>(null);
+
+  const [isScrollable, setIsScrollable] = useState(false);
+
+  /** When the selected tab changes, we need to check if the content is scrollable */
+  useEffect(() => {
+    setIsScrollable(
+      (scrollableContainerRef.current?.scrollHeight ?? 0) >
+        (scrollableContainerRef.current?.clientHeight ?? 0),
+    );
+  }, [selected, setIsScrollable]);
+
+  useEffect(() => {
+    if (!trackingEnabledSelector(store.getState())) {
+      return;
+    }
+    const data = {
+      button: `filter_${selected}`,
+    };
+    track(BUTTON_CLICKED_TRACK_EVENT, data);
+    braze.logCustomEvent(BUTTON_CLICKED_TRACK_EVENT, data);
```

Comment:
You should use segment not braze for this. You can use import { track } from "~/analytics";
trackingEnabledSelector not necessary we handle that directly when we track
With that changes you won't need to transform the import of braze in jest